pkg/controller/certificaterequests/vault/fuzz_test.go

```diff
@@ -0,0 +1,186 @@
+	"k8s.io/apimachinery/pkg/runtime"
+
+	internalinformers "github.com/cert-manager/cert-manager/internal/informers"
+	internalvault "github.com/cert-manager/cert-manager/internal/vault"
+	fakevault "github.com/cert-manager/cert-manager/internal/vault/fake"
+	apiutil "github.com/cert-manager/cert-manager/pkg/api/util"
+	"github.com/cert-manager/cert-manager/pkg/apis/certmanager"
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	controllerpkg "github.com/cert-manager/cert-manager/pkg/controller"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificaterequests"
+	testpkg "github.com/cert-manager/cert-manager/pkg/controller/test"
+	"github.com/cert-manager/cert-manager/pkg/util/pki"
+	"github.com/cert-manager/cert-manager/test/unit/gen"
+)
+
+var (
+	rsaSKFuzz *rsa.PrivateKey
+	err       error
+)
+
+func init() {
+	rsaSKFuzz, err = pki.GenerateRSAPrivateKey(2048)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func FuzzVaultSync(f *testing.F) {
+	f.Fuzz(func(t *testing.T,
+		secretTokenData,
+		customCsrPEM,
+		customRsaPEMCert []byte,
+		certDuration string,
+		addToken,
+		addCustomCsrPEM,
+		isCA bool,
+		baseCRCondition int) {
+		tm, err := time.ParseDuration(certDuration)
+		if err != nil {
+			return
+		}
+
+		// Add possibly invalid csrPEM or generate valid
+		var csrPEM []byte
+		if addCustomCsrPEM {
+			csrPEM = customCsrPEM
+		} else {
+			csrPEM = generateCSR(t, rsaSKFuzz)
+		}
+
+		fixedClockStart = time.Now()
+		metaFixedClockStart := metav1.NewTime(fixedClockStart)
+		baseIssuer := gen.Issuer("vault-issuer",
+			gen.SetIssuerVault(cmapi.VaultIssuer{
+				Server: "https://example.vault.com",
+			}),
+			gen.AddIssuerCondition(cmapi.IssuerCondition{
+				Type:   cmapi.IssuerConditionReady,
+				Status: cmmeta.ConditionTrue,
+			}),
+		)
+
+		baseCRNotApproved := gen.CertificateRequest("test-cr",
+			gen.SetCertificateRequestIsCA(isCA),
+			gen.SetCertificateRequestCSR(csrPEM),
+			gen.SetCertificateRequestDuration(&metav1.Duration{Duration: tm}),
+			gen.SetCertificateRequestIssuer(cmmeta.ObjectReference{
+				Name:  baseIssuer.Name,
+				Group: certmanager.GroupName,
+				Kind:  baseIssuer.Kind,
+			}),
+		)
+		var condition cmapi.CertificateRequestConditionType
+		switch baseCRCondition % 4 {
+		case 0:
+			condition = cmapi.CertificateRequestConditionReady
+		case 1:
+			condition = cmapi.CertificateRequestConditionInvalidRequest
+		case 2:
+			condition = cmapi.CertificateRequestConditionApproved
+		case 3:
+			condition = cmapi.CertificateRequestConditionDenied
+		}
+		baseCR := gen.CertificateRequestFrom(baseCRNotApproved,
+			gen.SetCertificateRequestStatusCondition(cmapi.CertificateRequestCondition{
+				Type:               condition,
+				Status:             cmmeta.ConditionTrue,
+				Reason:             "cert-manager.io",
+				Message:            "Certificate request has been approved by cert-manager.io",
```

Comment:
Can we change the message here to something generic (eg. "[test-message]"). Right now, it is a bit confusing that this message talks about approval while the condition can be Ready too.

Reply:
https://github.com/cert-manager/cert-manager/pull/7192/commits/08e14451ab102fe9ac02bbc1453a69214f248f7a